src/rpc/mempool.cpp

```diff
@@ -377,6 +427,63 @@ UniValue MempoolToJSON(const CTxMemPool& pool, bool verbose, bool include_mempoo
     }
 }
 
+static RPCHelpMan getmempoolfeesize()
+{
+    return RPCHelpMan{"getmempoolfeesize",
+        "Returns fee/size data for the whole mempool.",
+        {},
+        {},
+        RPCExamples{
+            HelpExampleCli("getmempoolfeesize", "")
+            + HelpExampleRpc("getmempoolfeesize", "")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+        {
+            const CTxMemPool& mempool = EnsureAnyMemPool(request.context);
+            LOCK(mempool.cs);
+
+            UniValue result(UniValue::VARR);
+
+            // TODO: refactor so that we're not just copying this from the miner.
+            std::vector<Cluster::HeapEntry> heap_chunks;
+            // Initialize the heap with the best entry from each cluster
+            for (const auto & [id, cluster] : mempool.m_cluster_map) {
+                if (!cluster->m_chunks.empty()) {
+                    heap_chunks.emplace_back(cluster->m_chunks.begin(), cluster.get());
+                }
+            }
+            // Define comparison operator on our heap entries (using feerate of chunks).
+            auto cmp = [](const Cluster::HeapEntry& a, const Cluster::HeapEntry& b) {
+                return a.first->fee*b.first->size < b.first->fee*a.first->size;
+            };
+            std::make_heap(heap_chunks.begin(), heap_chunks.end(), cmp);
+
+            CAmount accum_fee{0};
+            int64_t accum_size{0};
+            while (!heap_chunks.empty()) {
+                auto best_chunk = heap_chunks.front();
+                std::pop_heap(heap_chunks.begin(), heap_chunks.end(), cmp);
+                heap_chunks.pop_back();
+
+                accum_size += best_chunk.first->size;
+                accum_fee += best_chunk.first->fee;
+
+                UniValue o(UniValue::VOBJ);
```

Comment:
0b284b5fd29c06d46c1ec60ea7e1bcd07f36feb1: this should also list the cluster id, and maybe also the chunk number.

Could also add an argument limit the total number of vbytes.